frames/src/test/java/com/checkout/frames/component/cardholdername/CardHolderNameViewModelTest.kt

```diff
@@ -95,6 +103,86 @@ internal class CardHolderNameViewModelTest {
         assertEquals(viewModel.componentState.cardHolderName.value, filteredInput)
     }
 
+    /** Payment state related tests **/
+
+    @Test
+    fun `when cardHolderName updated then cardHolderName state in payment state manager is updated`() {
+        // Given
+        val testCardHolderName = "TestName"
+
+        // When
+        viewModel.onCardHolderNameChange(testCardHolderName)
+
+        // Then
+        assertEquals(spyPaymentStateManager.cardHolderName.value, testCardHolderName)
+    }
+
+    @Test
+    fun `when focus change triggered more than once then cardHolderName state in payment state manager is updated`() {
+        // Given
+        val testCardHolderName = "TestName"
+        viewModel.componentState.cardHolderName.value = testCardHolderName
+
+        // When
+        viewModel.onFocusChanged(true)
+        viewModel.onFocusChanged(false)
+
+        // Then
+        assertTrue(spyPaymentStateManager.isCardHolderNameValid.value)
+    }
+
+    /** Validation related tests **/
+
+    @Test
+    fun `when valid cardHolderName entered then error is hidden`() {
+        // Given
+        val testCardHolderName = "TestName"
+
+        // When
+        viewModel.onCardHolderNameChange(testCardHolderName)
+
+        // Then
+        with(viewModel.componentState.inputState) {
+            assertFalse(inputFieldState.isError.value)
+            assertFalse(errorState.isVisible.value)
+        }
+    }
+
+    @Test
+    fun `when cardHolderName is empty on focus change then error is shown`() {
+        // Given
```

Comment:
Shall we make the error state hidden and empty in `Given`? so that we can verify that it is shown as expected in `When`.